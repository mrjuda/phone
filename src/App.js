import React from 'react';
import {
  BrowserRouter, Routes, Route,
} from 'react-router-dom';
import Brands from './components/brands/Brands';
import Phones from './components/phones/Phones';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Brands />} />
          <Route path="/:slug" element={<Phones />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
