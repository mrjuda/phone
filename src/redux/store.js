import { configureStore, combineReducers } from '@reduxjs/toolkit';
import brandsSlice from './brands/brandsSlice';
import phonesSlice from './phones/phonesSlice';

const reducer = combineReducers({
  brands: brandsSlice.reducer,
  phones: phonesSlice.reducer,
});

const store = configureStore({ reducer });

export default store;
