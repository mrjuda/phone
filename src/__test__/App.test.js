// App.test.js

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import App from '../App';

describe(' Test App Rendering ', () => {
  it('App.js file renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
