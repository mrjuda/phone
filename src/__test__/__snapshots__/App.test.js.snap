// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test App Rendering  App.js file renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="brands-container"
  >
    <div
      className="header"
    >
      <div
        className="h-c1"
      >
        <img
          alt="logo"
          className="nav-icon"
          src="arrow-left.png"
        />
        <h2
          className="nav-text"
        >
          2022
        </h2>
      </div>
      <div
        className="h-c2"
      >
        <h1
          className="header-title"
        >
          devices by brand
        </h1>
      </div>
      <div
        className="h-c3"
      >
        <img
          alt="logo"
          className="nav-icon"
          src="mic.png"
        />
        <img
          alt="logo"
          className="nav-icon"
          src="gear.png"
        />
      </div>
    </div>
    <div
      className="banner"
    >
      DEVICES BY MAKER
    </div>
    <ul
      className="brands-list"
    />
  </div>
</div>
`;
